### DISCUSSION TITLE
Question/Request: Is there a microtonal and polyphonic sequencer or quantizer included in Cardinal?

### DISCUSSION THREAD

**Discussion by tristanbay** [DISTRHO/Cardinal#370] [Q&A]
I'm quite new to VCV Rack and was thinking of trying Cardinal, but before I download and try it, I wanted to ask if it has a module like [H4N4 Xen Quantizer](https://library.vcvrack.com/h4n4-modules/xen-qnt), which just quantizes signals polyphonically based on whatever Scala file is loaded into it. I also want to know if Cardinal has a sequencer that can generate notes from Scala files.

If you don't yet, would you be fine with adding the module I mentioned above into Cardinal? It's GPLv3+

Thanks

**Comment by dromer** [comment]
It might be better to convert this to a discussion so that people can discuss options.

But indeed this module could totally be added. However I do see one statement that is somewhat false:
`As bundled with this project it is sublicensed under the same license as the rest of the code, namely the GPL version 3 or later.`

That code is still MIT. People can use it however they want (also modify it and keep those modifications closed).
The author has no right to "sublicense" this module. (something to discuss with them)


Anyway for future reference there is a [discussion](https://github.com/DISTRHO/Cardinal/discussions/63) and [wiki page](https://github.com/DISTRHO/Cardinal/wiki/Possible-modules-to-include) on adding modules.
Please keep the issues for problems related to the project, not module requests.

**Reply by dromer** [reply]
@tristanbay a PR for adding the H4N4 module is open here -> https://github.com/DISTRHO/Cardinal/pull/349

**Reply by tristanbay** [reply]
@dromer
> a PR for adding the H4N4 module is open here -> #349

I noticed that you had already added the module to your own fork and submitted that PR to merge the code from there. I guess we wait for it to be accepted and included in a future Cardinal release now. You might have guessed it by now, but I'm still not that experienced with GitHub.

Also, semi unrelated question: Do you have any idea why there wasn't a Cardinal release last month?

**Reply by falkTX** [reply]
> Also, semi unrelated question: Do you have any idea why there wasn't a Cardinal release last month?

Sometimes people are just busy. Since I am mostly the one doing Cardinal dev work, and had to deal with some life things, there was no time for Cardinal. More details in https://www.patreon.com/posts/clarifications-70766698

But we are slowly coming back on track. 22.09 release seems possible

**Reply by dromer** [reply]
I think expectations that there will always be monthly releases are not realistic.
At some point the project will (should) settle down and updates will be less frequent.

Doesn't mean the project is dead of course, just that if things are stable and feature complete then updates are not really required either.
That and people also need time for their personal life and work for an income.

**Reply by falkTX** [reply]
yep. unlike some VCV folks we have work to do that is not Cardinal/Rack related.
Cardinal is just a side thing for fun, and always will be since we cant make it commercial anyhow
